src/lua/LuaPiGui.cpp

```diff
@@ -884,6 +884,17 @@ static int l_pigui_set_next_window_pos(lua_State *l)
 	return 0;
 }
 
+static int l_pigui_set_next_window_collapsed(lua_State *l)
+{
+	bool collapsed = true;
+	if (lua_isboolean(l, 1)) {
+		collapsed = lua_toboolean(l, 1);
+	}
```

Comment:
Note: this should be expressed as `bool collapsed = LuaPull<bool>(l, 1, true);` which does an implicit nil-check and handles the case in which the function stack is smaller than 1 (i.e. when the optional parameter is actually optional).

Reply:
Ahh, nice, I think I hadn't got my head around the `LuaPull` semantics at this time; that's much nicer.